vtex-apps/order-manager
# Updating orderForm in cart and checkout when necessary

#### What problem is this solving?

This pull request guarantees that the orderForm will always be up to date when the consumer gets to the cart/checkout pages.

#### How should this be manually tested?

[Here](https://refresh--checkoutio.myvtex.com/) add any item to the cart (don't close the page). Then, run [this](https://refresh--checkoutio.myvtex.com/_v/private/vtex.order-manager@0.9.0/graphiql/v1?query=mutation%20%7B%0A%20%20updateItems(orderItems%3A%20%7B%0A%20%20%20%20index%3A%200%2C%0A%20%20%20%20quantity%3A%203%0A%20%20%7D%2C%20allowedOutdatedData%3A%20%5B%22paymentData%22%5D)%20%7B%0A%20%20%20%20value%0A%20%20%20%20paymentData%7B%0A%20%20%20%20%20%20installmentOptions%7B%0A%20%20%20%20%20%20%20%20value%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D) mutation and check that the field `value` in `updateItems` differs from the `value` fields in `installmentOptions`. Then, on the first page, go to the cart or the checkout page. Finally, run [this](https://refresh--checkoutio.myvtex.com/_v/private/vtex.order-manager@0.9.0/graphiql/v1?query=query%20%7B%0A%20%20orderForm%7B%0A%20%20%20%20value%0A%20%20%20%20paymentData%7B%0A%20%20%20%20%20%20installmentOptions%7B%0A%20%20%20%20%20%20%20%20value%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D) query and check that the fields that were different are now equal. 

#### Checklist/Reminders


#### Screenshots or example usage

#### Type of changes


✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements 

#### Notes
